cub/test/catch2_test_device_histogram.cu

```diff
@@ -0,0 +1,652 @@
+      const auto* l = h_levels[channel].data();
+      const auto n  = static_cast<int>(h_levels[channel].size());
+      const int bin = static_cast<int>(
+        std::distance(l, std::upper_bound(l, l + n, static_cast<LevelT>(sample))) - 1); // TODO(bgruber):
+                                                                                        // why not
+                                                                                        // lower_bound?
+      return bin < 0 ? n : bin;
+    };
+    auto h_histogram = compute_reference_result<Channels, CounterT>(
+      h_samples, sample_to_bin_index, num_levels, width, height, row_pitch);
+
+    // Compute result and verify
+    {
+      const auto* sample_ptr = cast_if_half_pointer(thrust::raw_pointer_cast(d_samples.data()));
+      auto d_levels          = array<c2h::device_vector<LevelT>, ActiveChannels>{};
+      std::copy(h_levels.begin(), h_levels.end(), d_levels.begin());
+      _CCCL_IF_CONSTEXPR (ActiveChannels == 1 && Channels == 1)
+      {
+        histogram_range(
+          sample_ptr,
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_histogram[0].data())),
+          num_levels[0],
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_levels[0].data())),
+          width,
+          height,
+          row_pitch);
+      }
+      else
+      {
+        auto d_histogram_ptrs   = to_caller_vector_of_ptrs(d_histogram);
+        const auto d_num_levels = caller_vector<int>(num_levels.begin(), num_levels.end());
+        const auto level_ptrs   = to_caller_vector_of_ptrs(d_levels);
+        multi_histogram_range<Channels, ActiveChannels>(
+          sample_ptr,
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_histogram_ptrs.data())),
+          thrust::raw_pointer_cast(d_num_levels.data()),
+          cast_if_half_pointer(thrust::raw_pointer_cast(level_ptrs.data())),
+          width,
+          height,
+          row_pitch);
+      }
+    }
+    for (std::size_t c = 0; c < ActiveChannels; ++c)
+    {
+      CHECK(h_histogram[c] == d_histogram[c]);
+    }
+  }
+}
+
+using types =
+  c2h::type_list<std::int8_t,
+                 std::uint8_t,
+                 std::int16_t,
+                 std::uint16_t,
+                 std::int32_t,
+                 std::uint32_t,
+                 std::int64_t,
+                 std::uint64_t,
+#if TEST_HALF_T
+                 half_t,
+#endif
+                 float,
+                 double>;
+
+CUB_TEST("DeviceHistogram::Histogram* basic use", "[histogram][device]", types)
+{
+  using sample_t = c2h::get<0, TestType>;
+  using level_t =
+    typename std::conditional<cub::NumericTraits<sample_t>::CATEGORY == cub::FLOATING_POINT, sample_t, int>::type;
+  // Max for int8/uint8 is 2^8, for half_t is 2^10. Beyond, we would need a different level generation
+  const auto max_level       = level_t{sizeof(sample_t) == 1 ? 256 : 1024};
+  const auto max_level_count = (sizeof(sample_t) == 1 ? 256 : 1024) + 1;
```

Comment:
important: these two values are going to be used as an upper limit when generating samples. For both int8 and uint8, this leads to max level being equal to zero, and the sample being populated by zeroes. I'd preffer the sample be have more diverse values:
```suggestion
  const auto max_level       = level_t{sizeof(sample_t) == 1 ? 126 : 1024};
  const auto max_level_count = (sizeof(sample_t) == 1 ? 126 : 1024) + 1;
```

Reply:
Thx! I retained those values from the previous tests, but wondered myself.